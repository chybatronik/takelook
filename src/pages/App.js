import React, { Component } from 'react'
import { Layout, Input } from 'antd'
import { BrowserRouter as Router, Route } from 'react-router-dom'
import SearchPage from '../containers/Search'
import DetailsPage from '../containers/Details'
import DefaultListPage from '../containers/DefaultList'

import './App.css'
const Search = Input.Search
const { Header, Content, Footer } = Layout

class App extends Component {
  render () {
    return (
      <Router>
        <Layout>
          <Header style={{ position: 'fixed', width: '100%', zIndex: 1000 }}>
            <Search
              placeholder='Search Shows and People'
              onSearch={value => console.log(value)}
              enterButton
            />
          </Header>
          <Content style={{ padding: '0 50px', marginTop: 64, minHeight: '85vh' }}>
            <div style={{ margin: '16px 0', background: '#fff', padding: 24, minHeight: '80vh' }}>
              <Route exact path='/' component={DefaultListPage} />
              <Route path='/search/:name' component={SearchPage} />
              <Route path='/details/:id' component={DetailsPage} />
            </div>
          </Content>
          <Footer style={{ textAlign: 'center' }}>
            Takelook Â©2018 Created by chybatronik
          </Footer>
        </Layout>
      </Router>
    )
  }
}

export default App
