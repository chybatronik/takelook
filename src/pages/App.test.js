/**
 * @jest-environment node
 */
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import expect from 'expect'
import App from './App'

Enzyme.configure({ adapter: new Adapter() })

it('renders without crashing', () => {
  const output = shallow(
    <App />
  )
  expect(shallowToJson(output)).toMatchSnapshot()
})
