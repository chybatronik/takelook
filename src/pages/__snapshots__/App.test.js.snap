// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<BrowserRouter>
  <Adapter>
    <Adapter
      style={
        Object {
          "position": "fixed",
          "width": "100%",
          "zIndex": 1000,
        }
      }
    >
      <Search
        enterButton={true}
        inputPrefixCls="ant-input"
        onSearch={[Function]}
        placeholder="Search Shows and People"
        prefixCls="ant-input-search"
      />
    </Adapter>
    <Adapter
      style={
        Object {
          "marginTop": 64,
          "minHeight": "85vh",
          "padding": "0 50px",
        }
      }
    >
      <div
        style={
          Object {
            "background": "#fff",
            "margin": "16px 0",
            "minHeight": "80vh",
            "padding": 24,
          }
        }
      >
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/search/:name"
        />
        <Route
          component={[Function]}
          path="/details/:id"
        />
      </div>
    </Adapter>
    <Adapter
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Takelook Â©2018 Created by chybatronik
    </Adapter>
  </Adapter>
</BrowserRouter>
`;
